wp-includes/sqlite-ast/class-wp-sqlite-driver.php

```diff
@@ -1820,6 +2053,107 @@ private function translate_pure_identifier( WP_Parser_Node $node ): string {
 		return '`' . str_replace( '`', '``', $value ) . '`';
 	}
 
+	/**
+	 * Translate a qualified MySQL identifier to SQLite.
+	 *
+	 * The identifier can be composed of 1 to 3 parts (schema, object, child).
+	 *
+	 * @param  WP_Parser_Node|null $schema_node An identifier node representing a schema name (database).
+	 * @param  WP_Parser_Node|null $object_node An identifier node representing a database-level object name
+	 *                                          (table, view, procedure, trigger, etc.).
+	 * @param  WP_Parser_Node|null $child_node  An identifier node representing an object child name (column, index, etc.).
+	 * @return string                           The translated value.
+	 * @throws WP_SQLite_Driver_Exception       When the translation fails.
+	 */
+	private function translate_qualified_identifier(
+		?WP_Parser_Node $schema_node,
+		?WP_Parser_Node $object_node = null,
+		?WP_Parser_Node $child_node = null
+	): string {
+		$parts                = array();
+		$uses_reserved_prefix = false;
+
+		// Database name.
+		$is_information_schema = 'information_schema' === $this->db_name;
+		if ( null !== $schema_node ) {
+			$schema_name = $this->unquote_sqlite_identifier(
+				$this->translate_sequence( $schema_node->get_children() )
+			);
+			if ( 'information_schema' === strtolower( $schema_name ) ) {
+				$is_information_schema = true;
+			} elseif ( $this->db_name === $schema_name ) {
+				$is_information_schema = false;
+			} else {
+				throw $this->new_not_supported_exception(
+					sprintf(
+						"can't use schema '%s', only '%s' and 'information_schema' are supported",
+						$schema_name,
+						$this->db_name
+					)
+				);
+			}
+		}
+
+		/*
+		 * Make the 'information_schema' database read-only.
+		 *
+		 * This basic approach is rather restrictive, as it blocks the usage
+		 * of information schema tables in all data-modifying statements.
+		 *
+		 * Some of these statements can be valid, when the schema is only read:
+		 *   DELETE t FROM t JOIN information_schema.columns c ON ...
+		 *
+		 * If needed, a more granular approach can be implemented in the future.
+		 */
+		if ( true === $is_information_schema && false === $this->is_readonly ) {
```

Comment:
Good, let's block any writes to information_schema.

Reply:
Is there any way this identified would show up before we set `is_readonly`?

Reply:
@adamziel Not at the moment. The `$this->is_readonly = true` is set only on the top-level in the `execute_some_type_of_statement` methods. It's actually only SELECT, SHOW, DESCRIBE/EXPLAIN at the moment.